### DISCUSSION TITLE
AWS DynamoDB support for metadata store

### DISCUSSION THREAD

**Discussion by c3mb0** [juicedata/juicefs#1373] [Q&A] [answered]
Are the any plans to implement this? It seems like the use case would fit nicely unless I am missing something (maybe metadata range scan operations such as `ls *` cause a problem?)

**Comment by davies** [comment, marked as answer]
@c3mb0  JuiceFS relies on transactions API of database to ensure atomic operations,  after a quick look on the transactions [1] of DynamoDB, it looks like difficult to support DynamoDB.

For example, we need to get A, then find out key of B, and get BX, then update A and BX, they should be within single transaction, but is difficult to implement using just TransactWriteItems() and TransactGetItems()  

[1] https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/transaction-apis.html
